sktm/patchwork.py

```diff
@@ -622,18 +651,19 @@ def patchurl(self, patch):
         """
         return "%s/patch/%d" % (self.baseurl, patch.get("id"))
 
-    def log_patch(self, patch):
+    def log_patch(self, id, name, message_id, emails):
```

Comment:
I absolutely agree, it would be nicer. The problem is that some data it displays comes from mbox and some other data comes from the XML RPC patch object, so we cannot simply pass just the latter to it. I don't want to make some data type just for Patchwork v1 in a rush right now. Having this function with multiple arguments used only in two places in a single function in a single class kind-of lessens the impact, so I think it's OK.

The proper approach, I think, would be to implement the abstract patchwork interface class which would deal with a dedicated and well-described patch data type, which the specific patchwork implementations would work on retrieving. I'm willing to do that or a similar refactoring, but now we need to get this working. I can submit an issue proposing the refactoring and detailing things we can do.

Reply:
Makes sense, though if this will be used only in those two cases and won't rpovide any abstraction, do we need a special function for it? Anyways, please do submit the issue describing the problems with PW interfaces, it would be helpful to track so we have something to refer to later.

Reply:
Maybe we don't need it, or maybe we could use one after refactoring, we'll see. Submitted #38.